provider/shim/shim.go

```diff
@@ -2,14 +2,26 @@ package shim
 
 import (
 	"context"
+
+	pfprovider "github.com/hashicorp/terraform-plugin-framework/provider"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 	"github.com/hashicorp/terraform-provider-aws/internal/provider"
+	"github.com/hashicorp/terraform-provider-aws/internal/provider/fwprovider"
 )
 
-func NewProvider() *schema.Provider {
-	prov, err := provider.New(context.Background())
+type UpstreamProvider struct {
+	SDKV2Provider           *schema.Provider
+	PluginFrameworkProvider pfprovider.Provider
+}
+
+func NewUpstreamProvider(ctx context.Context) (UpstreamProvider, error) {
+	primary, err := provider.New(ctx)
 	if err != nil {
-		panic(err)
+		return UpstreamProvider{}, err
 	}
-	return prov
+	pf := fwprovider.New(primary)
+	return UpstreamProvider{
+		SDKV2Provider:           primary,
+		PluginFrameworkProvider: pf,
+	}, nil
```

Comment:
@t0yv0 I think the reason we are seeing behavior from the SDK side effect the PF side is because we are passing back the same provider (with its backing state).

Reply:
We have to do it. If we don't do it PF side is unable to deploy resources.

Reply:
Specifically it panics with malconfigured AWS client, which only gets configured in sdkv2-style callback. I can dig this up if it doesn't ring a bell.